controllers/apim/ratelimitpolicy_wasm_plugins.go

```diff
@@ -150,10 +151,10 @@ func (r *RateLimitPolicyReconciler) wasmPluginConfig(ctx context.Context,
 	if gwRLP != nil {
 		if len(gw.Hostnames()) == 0 {
 			// wildcard domain
-			gatewayActions["*"] = append(gatewayActions["*"], rlptools.GatewayActionsFromRateLimitPolicy(gwRLP)...)
+			gatewayActions["*"] = append(gatewayActions["*"], rlptools.GatewayActionsFromRateLimitPolicy(gwRLP, nil)...)
```

Comment:
Maybe we could refactor this `GatewayActionsFromRateLimitPolicy` now that it has some added functionality getting the rules from the `HTTPRoute` when a given _RLP_ has no rules... I was thinking that instead of using the second argument as a `route *gatewayapiv1alpha2.HTTPRoute`, we could use the networking object (Gw and Route) and extract the rules from there when no rules are defined and calling the fn the same way for both GW and HTTPRoute... that way maybe we could even get rid of `mergeGatewayActions` and having the logic within `wasmPluginConfig` a bit more descriptive.

Reply:
sorry, I do not follow. `GatewayActionsFromRateLimitPolicy` method is getting the rules from the HTTPRoute when the RLP has not rules. 

Can you elaborate more on that? surely, this `wasmPluginConfig` method can be improved for readability and maintainability.